flat_mutation_reader.cc

```diff
@@ -937,26 +941,39 @@ mutation_fragment_stream_validator::~mutation_fragment_stream_validator() {
 bool mutation_fragment_stream_validator::operator()(const mutation_fragment& mv) {
     auto kind = mv.mutation_fragment_kind();
     auto pos = mv.position();
+    auto region = pos.region();
     bool valid = false;
 
     fmr_logger.debug("[validator {}] {}:{}", this, kind, pos);
 
     if (mv.is_partition_start()) {
         valid = (_prev_kind == mutation_fragment::kind::partition_end);
-    } else {
+    } else if (_compare_keys) {
         auto less = position_in_partition::less_compare(_schema);
         if (_prev_kind != mutation_fragment::kind::range_tombstone) {
             valid = less(_prev_pos, pos);
         } else {
             valid = !less(pos, _prev_pos);
         }
+    } else if (region != partition_region::clustered) {
+        valid = (_prev_region < region);
+    } else {
+        valid = (_prev_region <= region);
     }
 
     if (__builtin_expect(!valid, false)) {
-        on_internal_error(fmr_logger, format("Unexpected mutation fragment: previous {}:{}, current {}:{}", _prev_kind, _prev_pos, kind, pos));
+        auto fmt = "Unexpected mutation fragment: previous {}:{}, current {}:{}";
```

Comment:
Missing `this` prefix for correlation with earlier log messages.

Reply:
ok, will fix in next version.